test/e2e/utils.go

```diff
@@ -0,0 +1,54 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/davecgh/go-spew/spew"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func updatedObjects(ctx context.Context, nsfilter string) ([]string, error) {
+	pods, err := clients.Kubernetes.CoreV1().Pods("openshift-azure-operator").List(ctx, metav1.ListOptions{
+		LabelSelector: "app=aro-operator-master",
+	})
+	if err != nil {
+		return nil, err
+	}
+	if len(pods.Items) != 1 {
+		return nil, fmt.Errorf("%d aro-operator-master pods found", len(pods.Items))
+	}
+	b, err := clients.Kubernetes.CoreV1().Pods("openshift-azure-operator").GetLogs(pods.Items[0].Name, &corev1.PodLogOptions{}).DoRaw(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	rx := regexp.MustCompile(`msg="(Update|Create) ([-a-zA-Z/.]+)`)
+	changes := rx.FindAllStringSubmatch(string(b), -1)
+	result := make([]string, 0, len(changes))
+	for _, change := range changes {
+		if nsfilter == "" || strings.Contains(change[2], "/"+nsfilter+"/") {
+			result = append(result, change[1]+" "+change[2])
+		}
+	}
+
+	return result, nil
+}
+
+func dumpEvents(ctx context.Context, namespace string) error {
+	events, err := clients.Kubernetes.EventsV1().Events(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, event := range events.Items {
+		spew.Dump(event)
```

Comment:
use a logger.

Reply:
Events data structure is not very friendly for this, this is easier :/ . This is just temporary code until we can identify flakes reason. I can migrate it but I dont see strong reasons for something like this which is just debug code until we get stuff sorted.